.github/workflows/test.yaml

```diff
@@ -0,0 +1,1552 @@
+          components: rustfmt, clippy
+      - name: Install LLVM (Linux)
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/clang+llvm-13.0.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz -L -o /opt/llvm.tar.xz
+          mkdir -p /opt/llvm-12
+          tar xf /opt/llvm.tar.xz --strip-components=1 -C /opt/llvm-12
+          echo '/opt/llvm-12/bin' >> $GITHUB_PATH
+          echo 'LLVM_SYS_120_PREFIX=/opt/llvm-12' >> $GITHUB_ENV
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+          shared-key: wasmer-linux-x64
+          save-if: false
+      - run: make lint
+        env:
+          ENABLE_CRANELIFT: "1"
+          ENABLE_LLVM: "1"
+          ENABLE_SINGLEPASS: "1"
+      - name: Assert no files have changed
+        run: |
+          git status
+          ! [[ $(git status -s) ]]
+
+  cargo_deny:
+    name: cargo-deny
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: EmbarkStudios/cargo-deny-action@v1
+
+  test_nodejs:
+    name: Test on NodeJS
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.63
+      - name: Install NodeJS
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - name: Install wasm-pack
+        run: |
+          curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh
+      - name: make test-js
+        run: |
+          make test-js
+      - name: make test-js-core
+        run: |
+          make test-js-core
+
+  test_wasm_build:
+    name: Test wasm build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          rustup target add wasm32-wasi
+          make build-wasmer-wasm
+
+  build_wasmer: 
+    name: Build wasmer on ${{ matrix.build }}
+    runs-on: ${{ matrix.os }}
+    needs: setup
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - build: linux-x64
+            os: ubuntu-18.04
+            target: x86_64-unknown-linux-gnu
+            llvm_url: 'https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/clang+llvm-13.0.0-x86_64-linux-gnu-ubuntu-16.04.tar.xz'
+            run_test: true
+            run_test_capi: true
+            run_integration_tests: true
+            run_build_docs: true
+            use_sccache: true
+          - build: macos-x64
+            os: macos-11
+            target: x86_64-apple-darwin
+            llvm_url: 'https://github.com/llvm/llvm-project/releases/download/llvmorg-13.0.0/clang+llvm-13.0.0-x86_64-apple-darwin.tar.xz'
+            run_test: true
+            run_test_capi: true
+            use_sccache: true
+            run_integration_tests: true
+            run_ios_tests: true
+            run_build_docs: false
+          - build: macos-x86_64
```

Comment:
This look like the same step as macos-x64. Maybe one should macos-arm64 instead?

Reply:
Yeah I think this was an error, I didn't see the duplicate `x86_64` when I experimented with runners.